tests/test_assignment.py

```diff
@@ -46,59 +46,59 @@ def test_submit_assignments(client, auth):
     response = client.get('/teacher/assignments')
     assert b'Biggerest' in response.data
-
-def test_create_assignments(client, auth, app):
+#Test the grading of the Assignments
+def test_grade_assignments(client, auth,app):
+    #Log in as teacher
     auth.teacher_login()
-
-    # On a GET request, user should see a form for creating new assignments
-    response = client.get('/teacher/assignments/create')
-    assert b'Assignment Creation' in response.data
-
-    # On POST, user should be able to insert new assignments into the database
-    client.post(
-        '/teacher/assignments/create',
-        data={'name': 'Wumbo Software', 'description': 'I wumbo, you wumbo, he, she, it... wumbo.',
-              'points': 900, 'course': 1}
-    )
+    #Response posts to this URL
+    response = client.get('/teacher/grade/submission')
+    # Assert the quest is redirected
+    assert response.status_code == 302
+    #Open up app-context
     with app.app_context():
+        #Assert that posting to the client makes you get redirected
+        assert client.get('/teacher/grade/submission', data={'grade-submission' : 100, 'submission': "(2, 1)"}).status_code == 302
+        #Send data to the form with grade as 100, for a student with id '2' for the 1st assignment of this session.
+        client.post('/teacher/grade/submission', data={'grade-submission' : 100, 'submission': "(2, 1)"})
+        # With database Select the first grade in the database, and check if the grade is 100
         with get_db() as con:
             with con.cursor() as cur:
-                cur.execute("SELECT * FROM assignments WHERE id = 5")
-                assert cur.fetchone()['name'] == 'Wumbo Software'
-
-def test_assign_work(client, auth):
-    auth.teacher_login()
-
-    # GET requests are redirected
-    response = client.get('/teacher/assignments/assign')
-    assert 'http://localhost/teacher/sessions' == response.headers['Location']
-
-    # On POST, user should be able to see assignments that belong to posted course id
-    # but only ones that have not been assigned already
-    response = client.post(
-        '/teacher/assignments/assign',
-        data={'session_id': 1}
-    )
-    assert b'Mondo Software' in response.data
-    assert b'Bigger Software' not in response.data
-
-def test_assign_submit(client, auth, app):
-    auth.teacher_login()
-
-    # GET requests are redirected
-    response = client.get('/teacher/assignments/assign/submit')
-    assert 'http://localhost/teacher/sessions' == response.headers['Location']
-
-    # On POST, user should be able to insert assignments into the session_assignments
-    # junction table with a due date
-    response = client.post(
-        '/teacher/assignments/assign/submit',
-        data={'date': '2020-05-08', 'session_id': 1, 'assign_id': 4}
-    )
-    with app.app_context():
+                cur.execute("""
+                SELECT * FROM assignment_grades
```

Comment:
Do you need to select everything if you only want one item from it?

Reply:
res[2] == '100' would make a lot more sense if we knew what index 2 actually is.